vms/platformvm/txs/executor/standard_tx_executor.go

```diff
@@ -959,6 +957,142 @@ func (e *standardTxExecutor) RegisterSubnetValidatorTx(tx *txs.RegisterSubnetVal
 	return nil
 }
 
+func (e *standardTxExecutor) SetSubnetValidatorWeightTx(tx *txs.SetSubnetValidatorWeightTx) error {
+	var (
+		currentTimestamp = e.state.GetTimestamp()
+		upgrades         = e.backend.Config.UpgradeConfig
+	)
+	if !upgrades.IsEtnaActivated(currentTimestamp) {
```

Comment:
I'm not super familiar with the code, but could the:
```
IsEtnaActivated()
SyntacticVerify()
VerifyMemoFieldLength()
CalculateFee()
VerifySpend()
```
flow be moved to a helper? Seems to be repeated (or at least close to repeated) for multiple tx types here.

Reply:
I do think this can be cleaned up... But I'd prefer it to be done holistically... Since this doesn't really impact correctness, I'm going to merge this PR and treat this as a followup.